crates/ruff_python_formatter/src/expression/expr_named_expr.rs

```diff
@@ -17,16 +18,37 @@ impl FormatNodeRule<ExprNamedExpr> for FormatExprNamedExpr {
             value,
             range: _,
         } = item;
-        write!(
-            f,
-            [
-                target.format(),
-                space(),
-                text(":="),
-                space(),
-                value.format(),
-            ]
-        )
+
+        write!(f, [target.format()])?;
+
+        let comments = f.context().comments().clone();
+        let trailing_target_comments = comments.trailing_comments(target.as_ref());
+        let leading_value_comments = comments.leading_comments(value.as_ref());
+        let dangling_item_comments = comments.dangling_comments(item);
+
+        if trailing_target_comments.is_empty() {
+            write!(f, [space()])?;
+        } else {
+            write!(f, [soft_line_break()])?;
+        }
+
+        write!(f, [text(":="), dangling_comments(dangling_item_comments)])?;
+
+        if leading_value_comments.is_empty() {
+            write!(f, [space()])?;
```

Comment:
What happens when there are dangling_item_comments but no leading leading_value_comments? this would make a good test case

Reply:
good catch. definitely something not what i expected. will continue investigating